### ISSUE TITLE
Question on how to generate 3d segmentation box results

### ISSUE THREAD

**Issue by apurvvarshney** [facebookresearch/egolifter#7]
![image](https://github.com/user-attachments/assets/5e07c375-8476-4c14-81fd-1d6ffae02354)
Hi, can you point me in the right direction on how to generate this view from your model, I am able to train and render my scene using your model.

Thanks!

**Comment by georgegu1997** [comment]
Hi, thanks for your interest!

To get such a visualization, we treated the mean of each 3D Gaussian as a point and performed HDBSCAN clustering on the point cloud (the distance metric for clustering is their distance in the learned contrastive feature space). 

The script we used for generating such visualizations can be found [here](https://github.com/facebookresearch/egolifter/blob/vis_pcd/vis_pcd.py). This script was used with an older version of the codebase and it needs some (minor) adaptations to work. But this script should describe the overall process we used for the visualization well. Feel free to reply if you find it working or not, or make a PR if you get it working!
